packages/commerce-sdk-react/src/hooks/ShopperBaskets/useShopperBasketAction.ts

```diff
@@ -0,0 +1,15 @@
+import {ActionResponse} from '../../types'
+import {ShopperBasketActions} from './types'
+
+// Q: how do we link the return types for actions?
+type execute = () => Promise<any>
+
+const useShopperBasketAction = (action: ShopperBasketActions): ActionResponse<execute> => {
```

Comment:
Are we planning on defining each action in a separate file under an `/action` folder?

If not, based on the list of `ShopperBasketActions` the file could be too long.

https://github.com/SalesforceCommerceCloud/pwa-kit/blob/54bd7b5b9538a112a09d0fab2326e0cf5ef2dd74/packages/commerce-sdk-react/src/hooks/ShopperBaskets/types.ts#L8-L35

Reply:
Good question and I think we will know the answer when we implement the very first action hook. I'd like to defer this to the next ticket.